simulation/agents.py

```diff
@@ -437,19 +437,19 @@ def get_total_heating_system_costs(
             heating_system, model
         )
 
-        if model.intervention == InterventionType.BOILER_UPGRADE_SCHEME:
+        if InterventionType.BOILER_UPGRADE_SCHEME in model.interventions:
             subsidies = estimate_boiler_upgrade_scheme_grant(heating_system, model)
             if subsidies > 0:
                 self.boiler_upgrade_grant_available = True
 
-        if model.intervention == InterventionType.RHI:
+        elif InterventionType.RHI in model.interventions:
             rhi_annual_payment = estimate_rhi_annual_payment(self, heating_system)
             subsidies = discount_annual_cash_flow(
```

Comment:
I think this is fine for our use case, because all the scenarios we want to run won't have conflicting subsidies where the agents will have to "choose which subsidy applies to them", right?

Technically in the current setup you could run RHI and BUS at the same time, but because they apply over non-intersecting time windows I don't think it matters. 

Also we were only using RHI for calibration. In our actual scenario setup we will run {Boiler Ban, BUS, BUS + Policy Costs, BUS + Policy Costs + Boiler Ban}. I think, handily, none of these are conflicting in terms of competing subsidies: in the latter scenario the first intervention impacts subsidies, the second intervention impacts running costs and the third intervention impacts heating system availability.

So I think this implementation is OK, but just wanted a comment here to check you agree there wouldn't be any weird behaviour?

Reply:
Exactly! Upfront subsidies only apply for the RHI and BUS, which happen to be mutually exclusive.